Sources/PostgreSQL/Connection.swift

```diff
@@ -171,6 +180,37 @@ public final class Connection: ConnInfoInitializable {
         }
     }
     
+    /// Creates a dispatch read source for this connection that will call `callback` on `queue` when a notification is received
+    ///
+    /// - Parameter channel: the channel to register for
+    /// - Parameter queue: the queue to create the DispatchSource on
+    /// - Parameter callback: the callback
+    /// - Parameter note: The notification received from the database
+    /// - Parameter err: Any error while reading the notification. If not nil, the source will have been canceled
+    /// - Returns: the dispatch socket to activate
+    /// - Throws: if fails to get the socket for the connection
+    public func makeListenDispatchSource(toChannel channel: String, queue: DispatchQueue, callback: @escaping (_ note: Notification?, _ err: Error?) -> Void) throws -> DispatchSourceRead {
+        guard let sock = Optional.some(PQsocket(self.cConnection)), sock >= 0
```

Comment:
This seems a bit strange, why create an optional out of something not optional. Also, there is a callback, I would use this callback in case of an error. This way you can remove `throws` from this function. So it would change to:

```swift
let sock = PQsocket(cConnection)
guard sock >= 0 else {
    let error = PostgreSQLError(code: .ioError, reason: "failed to get socket for connection")
    callback(nil, error)
    return
}
```

Reply:
The optional allows using it in a guard statement. I try to put all precondition guards at the top of the function.  I'll change. 

It can't just return because it returns a non-optional DispatchSource. That's why it throws. Better to explicitly generate an error instead of returning nil.